avr-hal-generic/src/spi.rs

```diff
@@ -68,13 +72,17 @@ impl Default for Settings {
 /// This trait defines the common interface for all SPI peripheral variants.  It is used as an
 /// intermediate abstraction ontop of which the [`Spi`] API is built.  **Prefer using the
 /// [`Spi`] API instead of this trait.**
-pub trait SpiOps<H, SCLK, MOSI, MISO, CS> {
+pub trait SpiOps<H, SCLK, MOSI, MISO> {
```

Comment:
You can't drop the CS pin here, as it has meaning to the hardware.  You can read up on the details in https://github.com/Rahix/avr-hal/issues/27.  We had to add this rather inflexible CS abstraction to ensure you can't cause unintentional side effects by misusing the CS pin.